src/lex/cpp.rs

```diff
@@ -1644,4 +1644,31 @@ int main(){}
         );
         // TODO test for #includes
     }
+
+    fn assert_same_stringified(cpp: &str, cpp_src: &str) {
+        assert_same_exact(
+            &format!("#define xstr(a) #a\nxstr({})", cpp),
+            &format!("\n{}", cpp_src),
+        );
+    }
+    #[test]
+    fn stringify() {
+        assert_same_stringified("a + b", r#""a + b""#);
+        assert_same_stringified(r#"b   	 +"c""#, r#""b +\"c\"""#);
+        assert_same_stringified(r#""+\\+\n+\"+""#, r#""\"+\\\\+\\n+\\\"+\"""#);
+        assert_same_exact("#define xstr(a, b) #a = b\nxstr(1+2,3)", "\n\"1+2\" = 3");
+        assert_same_exact("#define xstr(a, b) a b\nxstr(1+2,3)", "\n1+2 3");
+        assert_same_exact("#define xstr(a) # a\nxstr(1+2)", "\n\"1+2\"");
+        assert_same_exact("#define hash #a\nhash", "\n#a");
+    }
+    #[test]
+    #[ignore = "Depends on #384"]
+    fn stringify_string() {
```

Comment:
These tests can be brought in after #384 is implemented.

Reply:
Can you change `ignore` to `should_panic`? That way we'll have to update the test when it's been fixed.

Reply:
It doesn't panic but just gives the wrong output.

Reply:
Ok, this is fine for now then.